components-core/src/main/java/com/airwallex/android/core/ClientSecretRepository.kt

```diff
@@ -23,6 +23,17 @@ class ClientSecretRepository(private val provider: ClientSecretProvider) {
         }
     }
 
+    @Suppress("SwallowedException", "TooGenericExceptionThrown", "TooGenericExceptionCaught")
+    suspend fun retrieveClientSecret(customerId: String): ClientSecret {
+        return withContext(Dispatchers.IO) {
+            try {
+                provider.provideClientSecret(customerId)
+            } catch (e: Exception) {
+                throw Exception("Could not retrieve client secret from the given provider")
```

Comment:
Might be good to make a specific exception for this to avoid the `TooGenericExceptionThrown` suppression, and just so the frontend can handle it differently if it chooses

Reply:
Sure, I'll throw it as a `AirwallexCheckoutException`